### PR TITLE
Optimize SpEL and property placeholder support for @Async qualifiers

### PR DESCRIPTION
**Pull request by dawn9117**
optimize spel in AsyncExecutionAspectSupport

### CODE REVIEW

**Comment by dawn9117** [comment]
> spel is already supported.

Okay, but such as @Async (" ${executor.name:} ") doesn't work  when executor.name not configured.
Additionally EmbeddedValueResolver can be reused.

**Comment by sbrannen** [comment]
### Related Issues

-  #27818

File: spring-aop/src/main/java/org/springframework/aop/interceptor/AsyncExecutionAspectSupport.java [hunk 1]
```diff
@@ -167,6 +174,9 @@ protected AsyncTaskExecutor determineAsyncExecutor(Method method) {
 		if (executor == null) {
 			Executor targetExecutor;
 			String qualifier = getExecutorQualifier(method);
+			if (stringValueResolver != null && StringUtils.hasLength(qualifier)) {
+				qualifier = stringValueResolver.resolveStringValue(qualifier);
```
Comment (sbrannen) [inline spring-aop/src/main/java/org/springframework/aop/interceptor/AsyncExecutionAspectSupport.java:178]: This change currently [fails the build](https://ci.spring.io/builds/190047#L62728b6d:254:256).

```suggestion
			if (this.stringValueResolver != null && StringUtils.hasLength(qualifier)) {
				qualifier = this.stringValueResolver.resolveStringValue(qualifier);
```

**Review by sbrannen — CHANGES_REQUESTED** [review]
Please make the requested change and introduce a test that fails before this PR and passes after this PR.

**Comment by dawn9117** [comment]
> Please make the requested change and introduce a test that fails before this PR and passes after this PR.

I've already added it. Please review, thanks.

**Comment by sbrannen** [comment]
> I've already added it. Please review, thanks.

Thanks for making the requested changes. I'll see about getting it merged in.

**Review by sbrannen — APPROVED** [review]

**Comment by sbrannen** [comment]
This has been merged into `main`.

Thanks 👍

**Pull request closed without merging**